ncm-ceph/src/main/perl/Ceph/ClusterMap.pm

```diff
@@ -0,0 +1,229 @@
+#${PMpre} NCM::Component::Ceph::ClusterMap${PMpost}
+
+use parent qw(CAF::Object);
+use Readonly;
+use Data::Dumper;
+use JSON::XS;
+
+
+sub _initialize
+{
+    my ($self, $clusterobj) = @_;
+
+    $self->{log} = $clusterobj;
+
+    $self->{quattor} = {};
+    $self->{ceph} = {};
+
+    $self->{Cluster} = $clusterobj;
+    $self->{deploy} = {};
+    return 1;
+}
+
+sub get_fqdn
+{
+    my ($self, $host) = @_;
+    return $self->{quattor}->{$host}->{fqdn};
+};
+
+# get the parsed daemons as they are declared in schema
+sub get_quattor_map
+{
+    my ($self) = @_;
+    return $self->{quattor};
+}
+
+# Gets the MON map
+sub mon_hash
+{
+    my ($self) = @_;
+    my ($ec, $jstr) = $self->{Cluster}->run_ceph_command([qw(mon dump)], 'get mon map', nostderr => 1) or return;
+    my $monsh = decode_json($jstr);
+    foreach my $mon (@{$monsh->{mons}}){
+        $self->add_existing('mon', $mon->{name}, { addr => $mon->{addr}});
+    }
+    return 1;
+}
+
+# Gets the MGR map
+sub mgr_hash
+{
+    my ($self) = @_;
+    my ($ec, $jstr) = $self->{Cluster}->run_ceph_command([qw(mgr dump)], 'get mgr map', nostderr => 1) or return;
+    my $mgrsh = decode_json($jstr);
+    $self->add_existing('mgr', $mgrsh->{active_name});
+    foreach my $mgr (@{$mgrsh->{standbys}}){
+        $self->add_existing('mgr', $mgr->{name});
+    }
+    return 1;
+}
+
+# Gets the MDS map
+sub mds_hash
+{
+    my ($self) = @_;
+    my ($ec, $jstr) = $self->{Cluster}->run_ceph_command([qw(mds stat)], 'get mds map', nostderr => 1) or return;
+    my $mdshs = decode_json($jstr);
+    my $fsmap = $mdshs->{fsmap};
+    foreach my $fs (@{$fsmap->{filesystems}}){
+        foreach my $mds (values %{$fs->{mdsmap}->{info}}) {
+            $self->add_existing('mds', $mds->{name});
+        }
+    }
+    foreach my $mds (@{$fsmap->{standbys}}){
+        $self->add_existing('mds', $mds->{name});
+    }
+
+    return 1;
+}
+
+# Compare and change mon config
+sub check_mon
+{
+    my ($self, $hostname, $mon) = @_;
+    $self->debug(3, "Comparing mon $hostname");
+    my $ceph_mon = $self->{ceph}->{$hostname}->{mon};
+    if ($ceph_mon->{addr} =~ /^0\.0\.0\.0:0/) {
```

Comment:
use `eq`?

Reply:
there's no othe rway to check this?

Reply:
This is a corner case seen with an earlier version. running the ceph-deploy create initial command sometimes went wrong and the ceph-mon is in a weird existing not configured state. Don't know if this can still happen or how to reproduce this

Reply:
please add this in a comment